### PR TITLE
Compatibility with Pydantic >= 2 through pydantic.v1

### PR DESCRIPTION
**Pull request by SarahG-579462**
# Overview

Recently, pygeoapi's pydantic was downgraded from 2.0-compatible to pydantic < 1. This removed compatibility with two of our libraries, [xscen](https://github.com/Ouranosinc/xscen) and [xclim](https://github.com/Ouranosinc/xclim), for climate data processing, due to [intake-esm's current requirement on pydantic > 2](https://github.com/intake/intake-esm/blob/60766304f86173d8cad970e8979693e281f80658/requirements.txt#L10)

Because we require these in the use of our pygeoapi process plugins, this change forced us to use either very outdated versions of xscen (and thus xarray, pandas, xclim, etc.), or a very outdated version of pygeoapi.

Thankfully, a simple fix is available by the use of pydantic 2's backwards-compatible pydantic.v1 module. 

It is not entirely clear to me why pydantic was downgraded, the best explanation I can see in the discussion was due to "versionitis". I was, however, able to run all tests with the latest version of pydantic installed in this fashion. The way this PR is written, if pydantic >= 2 is not available on the operating system in question, everything will continue as-is.

# Related Issue / discussion

Pydantic issues:
#1573 
#1341
#1584

# Additional information

# Dependency policy (RFC2)

- [x] I have ensured that this PR meets [RFC2](https://pygeoapi.io/development/rfc/2) requirements

(It follows RFC2 as far as I understand, as it allows backwards compatibility with pydantic < 2)

# Updates to public demo

- [ ] I have ensured that breaking changes to the [pygeoapi master demo server](https://github.com/geopython/demo.pygeoapi.io) have been addressed
  - [ ] https://github.com/geopython/demo.pygeoapi.io/blob/master/services/pygeoapi_master/local.config.yml

No breaking changes shuold ensue as a result of this PR, since it is backwards compatible. I am not entirely sure how to test this demo server, help would be appreciated here.

# Contributions and licensing

(as per https://github.com/geopython/pygeoapi/blob/master/CONTRIBUTING.md#contributions-and-licensing)

- [x] I'd like to contribute [feature X|bugfix Y|docs|something else] to pygeoapi. I confirm that my contributions to pygeoapi will be compatible with the pygeoapi license guidelines at the time of contribution
- [x] I have already previously agreed to the pygeoapi Contributions and Licensing Guidelines

### CODE REVIEW

File: .github/workflows/vulnerabilities.yml [hunk 1]
```diff
@@ -21,6 +21,10 @@ jobs:
       run:
         working-directory: .
     steps:
+    - id: string
```
Comment (tomkralidis) [inline .github/workflows/vulnerabilities.yml:24]: What is the nature of this update?  Should this be in another PR?

File: requirements-django.txt [hunk 2]
```diff
@@ -1,2 +1,2 @@
 Django
-pydantic<2.0
+pydantic
```
Comment (tomkralidis) [inline requirements-django.txt:2]: Not sure how this landed in `requirements-django.txt`, but given that pydantic is in `requirements.txt` I would actually remove this requirement in `requirements-django.txt`.

File: requirements.txt [hunk 3]
```diff
@@ -4,7 +4,7 @@ filelock
 Flask
 jinja2
 jsonschema
-pydantic<2.0
+pydantic
```
Comment (tomkralidis) [inline requirements.txt:7]: Is there a timeline for pydantic 2 on Ubuntu 22.04?  We need to consider this for this PR (note that our next release is scheduled for next month).

**Review by tomkralidis — CHANGES_REQUESTED** [review]

File: .github/workflows/vulnerabilities.yml (hunk 1 above)
Comment (SarahG-579462) [reply .github/workflows/vulnerabilities.yml:24]: This is related to the comment above. because I was working on a fork which has a capital letter in the repo name, it was failing to build the docker image in this github action. I don't think it's a great idea for this particular case, because it relies on a third party github action for security vulnerability checks, so I'll yeet it. :slightly_smiling_face:

File: requirements.txt (hunk 3 above)
Comment (SarahG-579462) [reply requirements.txt:7]: Are you saying that without `pydantic<2.0` here, it will override the system pydantic version, if someone is building without a virtual env, and that is an issue?

**Comment by tomkralidis** [comment]
To add, it would be valuable to have pygeoapi.models.pydantic that covers the try/except in one place and have the rest of code import from pygeoapi.models.pydantic

**Comment by tomkralidis** [comment]
Closing given https://github.com/geopython/pygeoapi/issues/1664#issuecomment-2192417748
